java/sage/FFMPEGTranscoder.java

```diff
@@ -781,14 +782,22 @@ public void startTranscode() throws java.io.IOException
       xcodeParamsVec.add("mpegts");
       xcodeParamsVec.add("-vcodec");
       xcodeParamsVec.add(videoCodec = "libx264");
-      targetWidth = 480;
-      targetHeight = 272;
+      String sizeKey = String.format(BITRATE_OPTIONS_SIZE_KEY, estimatedBandwidth/1000);
+      String xcodeSize = Sage.get(sizeKey, Sage.get(String.format(BITRATE_OPTIONS_SIZE_KEY, "default"), "480x272"));
+      if (Sage.DBG)
+        System.out.println("FFMpegTranscoder: httpls: Using framesize "+xcodeSize+" for bandwidth: "+(estimatedBandwidth/1000)+" base on key: " + sizeKey);
+      // this will always return a valid 2 element array of w and h
+      int size[] = parseFrameSize(xcodeSize, 480, 272);
+      if (Sage.DBG)
+        System.out.println("FFMpegTranscoder: httpls: Calculated framesize " + size[0] + "x" + size[1]);
+      targetWidth = size[0];
+      targetHeight = size[1];
       currAudioBitrateKbps = 32;
       currVideoBitrateKbps = (int)Math.max(64000, (estimatedBandwidth - 32000))/1000;
       xcodeParamsVec.add("-b");
       xcodeParamsVec.add(currVideoBitrateKbps*1000 + "");
       xcodeParamsVec.add("-s");
-      xcodeParamsVec.add("480x272");
+      xcodeParamsVec.add(size[0] + "x" + size[1]);
```

Comment:
I'd do this for better clarity:
xocdeParamsVec.add(targetWidth + "x" + targetHeight);